File: Problem_1695/solution.py
from typing import List


class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        result = nums[0]
        
        left = 0
        sum = nums[0]
        myset = set([nums[0]])
        for i in range(1, len(nums)):
            while (nums[i] in myset and left < i):
                myset.remove(nums[left])
                sum -= nums[left]
                left += 1
            sum += nums[i]
            myset.add(nums[i])
            if (sum > result):
                result = sum
            # print(i, result, left)
        return result
    
s = Solution()
# print(s.maximumUniqueSubbarray([4,2,4,5,6]))
print(s.maximumUniqueSubbarray([5,2,1,2,5,2,1,2,5]))
            

File: Problem_1695/solution.cs
namespace LeetCode;
public class Solution1695
{
    public int MaximumUniqueSubarray(int[] nums)
    {
        HashSet<int> found = new();
        int max = 0;
        int sum = 0;
        int start = 0;
        for (int i = 0; i < nums.Length; i++)
        {
            if (found.Contains(nums[i]))
            {
                for (; nums[start] != nums[i]; start++)
                {
                    sum -= nums[start];
                    found.Remove(nums[start]);
                }
                start++;
            }
            else
            {
                sum += nums[i];
                max = Math.Max(sum, max);
                found.Add(nums[i]);
            }
        }
        return max;
    }
}